src/textual/widgets/_input.py

```diff
@@ -571,13 +571,14 @@ def set_classes() -> None:
             self.set_class(not valid, "-invalid")
             self.set_class(valid, "-valid")
 
-        # If no validators are supplied, and therefore no validation occurs, we return None.
-        if not self.validators:
+        # Empty inputs are always OK if valid_empty is set.
+        if self.valid_empty and not value:
             self._valid = True
             set_classes()
             return None
 
-        if self.valid_empty and not value:
+        # If no validators are supplied, and therefore no validation occurs, we return None.
+        if not self.validators:
```

Comment:
Since we here have two conditions with the same action, would it overcomplicate the code to elide them? At present I suspect the sequence improves readability.